fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -1471,18 +1472,43 @@ private String mapToActualAppTable(final String appTable) {
         return appTable;
     }
 
-    private List<ResultsetRowData> fillDatatableResultSetDataRows(final String sql) {
+    private List<ResultsetRowData> fillDatatableResultSetDataRows(final String sql, final List<ResultsetColumnHeaderData> columnHeaders) {
         final List<ResultsetRowData> resultsetDataRows = new ArrayList<>();
+        final GenericResultsetData genericResultsetData = new GenericResultsetData(columnHeaders, null);
 
         final SqlRowSet rowSet = jdbcTemplate.queryForRowSet(sql); // NOSONAR
         final SqlRowSetMetaData rsmd = rowSet.getMetaData();
 
         while (rowSet.next()) {
-            final List<String> columnValues = new ArrayList<>();
+            final List<Object> columnValues = new ArrayList<>();
+
             for (int i = 0; i < rsmd.getColumnCount(); i++) {
                 final String columnName = rsmd.getColumnName(i + 1);
-                final String columnValue = rowSet.getString(columnName);
-                columnValues.add(columnValue);
+                if (columnHeaders.isEmpty()) {
+                    columnValues.add(rowSet.getString(columnName));
+                } else {
+                    final String colType = genericResultsetData.getColTypeOfColumnNamed(columnName);
```

Comment:
This looks still over complexed.

Something like this is enough, i believe:

`while (rowSet.next()) {
            final List<Object> columnValues = new ArrayList<>();

            for (int i = 0; i < rsmd.getColumnCount(); i++) {
                final String columnName = rsmd.getColumnName(i + 1);
                final String colType = genericResultsetData.getColTypeOfColumnNamed(columnName);
                if ("DECIMAL".equalsIgnoreCase(colType)) {
                    columnValues.add(rowSet.getBigDecimal(columnName));
                } else if ("DATE".equalsIgnoreCase(colType) || ("TIMESTAMP WITHOUT TIME ZONE".equalsIgnoreCase(colType) // PostgreSQL
                        || "DATETIME".equalsIgnoreCase(colType) || "TIMESTAMP".equalsIgnoreCase(colType))) {
                    columnValues.add(rowSet.getObject(columnName));
                } else {
                    columnValues.add(rowSet.getString(columnName));
                }
            }

            resultsetDataRows.add(ResultsetRowData.create(columnValues));
        }`

Reply:
Done! Updated

<img width="1130" alt="Screen Shot 2022-07-29 at 9 32 17" src="https://user-images.githubusercontent.com/44206706/181782554-47a31294-c00b-471e-97fe-60f575add40b.png">